<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>VisualizeSort</class>
 <widget class="QMainWindow" name="VisualizeSort">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>796</width>
    <height>363</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>VisualizeSort</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QWidget" name="canvasPanel" native="true">
      <property name="styleSheet">
       <string notr="true"/>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QWidget" name="controlPanel" native="true">
      <property name="minimumSize">
       <size>
        <width>250</width>
        <height>300</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>250</width>
        <height>16777215</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="meansSelect">
         <item>
          <widget class="QLabel" name="meansLabel">
           <property name="font">
            <font>
             <family>宋体</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="text">
            <string>排序方法：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="meansCombo">
           <property name="font">
            <font>
             <family>宋体</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="editable">
            <bool>false</bool>
           </property>
           <property name="maxVisibleItems">
            <number>10</number>
           </property>
           <item>
            <property name="text">
             <string>冒泡排序</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>优化的冒泡排序</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>鸡尾酒排序</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>快速排序</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>直接插入排序</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>二分插入排序</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>希尔排序</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>计数排序</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>猴子排序</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>基数排序</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>归并排序</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="number">
         <item>
          <widget class="QLabel" name="numberLabel">
           <property name="font">
            <font>
             <family>宋体</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="text">
            <string>数据个数：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="numberSpin">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="value">
            <number>20</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="sleepTime">
         <item>
          <widget class="QLabel" name="timeLabel">
           <property name="font">
            <font>
             <family>宋体</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="text">
            <string>等待时间：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="timeSpin">
           <property name="font">
            <font>
             <family>Times New Roman</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="maximum">
            <number>999999999</number>
           </property>
           <property name="value">
            <number>200</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="controlButton">
         <item>
          <widget class="QPushButton" name="startButton">
           <property name="font">
            <font>
             <family>宋体</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="text">
            <string>开始排序</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="initializeButton">
           <property name="font">
            <font>
             <family>宋体</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="text">
            <string>初始化数据</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>796</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
